runtime/runtime/src/lib.rs

```diff
@@ -1286,17 +1303,35 @@ impl Runtime {
         // the check is not necessary.  It’s defence in depth to make sure any
         // future refactoring won’t break the condition.
         assert!(cfg!(feature = "sandbox") || state_patch.is_empty());
-
         let protocol_version = apply_state.current_protocol_version;
-        let _span = tracing::debug_span!(
-            target: "runtime",
-            "apply",
-            protocol_version,
-            num_transactions = transactions.len())
-        .entered();
-
         let mut prefetcher = TriePrefetcher::new_if_enabled(&trie);
         let mut state_update = TrieUpdate::new(trie);
+        struct TotalResourceGuard {
```

Comment:
I'll have to think if this is the best place for this struct definition. It feels to me that it is -- it is only used within this function. On the other hand this function is humongous and this struct definition right here ain't helping with that problem.